python_modules/dagstermill/dagstermill/__init__.py

```diff
@@ -132,21 +137,39 @@ def yield_result(self, value, output_name):
         if not self.populated_by_papermill:
             return value
 
-        check.invariant(
-            self.solid_def is not None,
-            "If Dagstermill has been run by papermill, self.solid_def should not be None",
-        )
-        if not self.solid_def.has_output(output_name):
-            raise DagstermillError(
-                'Solid {solid_name} does not have output named {output_name}'.format(
-                    solid_name=self.solid_def.name, output_name=output_name
+        if self.solid_def is None:
+            if output_name not in self.output_name_type_dict:
+                raise DagstermillError(
+                    'Solid {solid_name} does not have output named {output_name}'.format(
+                        solid_name=self.solid_def_name, output_name=output_name
+                    )
+                )
+            runtime_type_enum = self.output_name_type_dict[output_name]
+            if runtime_type_enum == SerializableRuntimeType.SCALAR:
+                pm.record(output_name, value)
+            elif runtime_type_enum == SerializableRuntimeType.ANY and is_json_serializable(value):
+                pm.record(output_name, value)
+            elif runtime_type_enum == SerializableRuntimeType.PICKLE_SERIALIZATION_STRATEGY:
```

Comment:
ok. this is a little strange since `SCALAR`, `ANY`, and `PICKLE_SERIALIZATION_STRATEGY` aren't obviously the same kind of thing. `PICKLE_SERIALIZABLE`?

Reply:
changed it